include/services/internal/collection.h

```diff
@@ -28,8 +29,112 @@ namespace services
 {
 namespace internal
 {
+
+/**
+ * <a name="DAAL-CLASS-SERVICES__INTERNAL__PRIMITIVECOLLECTION"></a>
+ * \brief  Class that provides simple memory management routines for handling blocks
+ *         of continues memory, also provides automatic memory deallocation. Note this
+ *         class doesn't provide functionality for objects constructions and simply allocates
+ *         and deallocates memory. In case of objects consider Collection or ObjectPtrCollection
+ * \tparam T Type of elements which are stored in the buffer
+ */
+template<typename T>
+class PrimitiveCollection : public Base
+{
+public:
+    PrimitiveCollection() :
+        _buffer(NULL),
+        _size(0) { }
+
+    explicit PrimitiveCollection(size_t size, services::Status *status = NULL)
+    {
+        services::Status localStatus = reallocate(size);
+        services::internal::tryAssignStatusAndThrow(status, localStatus);
+    }
+
+    virtual ~PrimitiveCollection()
+    {
+        destroy();
+    }
+
+    void destroy()
+    {
+        services::daal_free((void *)_buffer);
+        _buffer = NULL;
```

Comment:
nullptr is used in other parts of DAAL

Reply:
This is not SYCL part of the library, C++98 is required

Reply:
Ok